modules/ollama/testcontainers/ollama/__init__.py

```diff
@@ -0,0 +1,76 @@
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import docker
+
+from testcontainers.core.generic import ServerContainer
+from testcontainers.core.waiting_utils import wait_container_is_ready
+
+
+class OllamaContainer(ServerContainer):
+    def __init__(self, image="ollama/ollama:latest"):
+        self.port = 11434
+        super().__init__(port=self.port, image=image)
+        self.with_exposed_ports(self.port)
+        self._check_and_add_gpu_capabilities()
+
+    def _check_and_add_gpu_capabilities(self):
+        info = docker.APIClient().info()
+        if "nvidia" in info["Runtimes"]:
+            self.with_create_container_kwargs(
+                {"HostConfig": {"DeviceRequests": [{"Capabilities": [["gpu"]], "Count": -1}]}}
+            )
+
+    def start(self):
+        """
+        Start the Ollama server
+        """
+        super().start()
+        wait_container_is_ready(self, "Ollama started successfully")
+        self._connect()
+
+        return self
+
+    def get_endpoint(self):
+        """
+        Return the endpoint of the Ollama server
+        """
+        return self._create_connection_url()
+
+    def get_container(self):
+        """
+        Return the container object
+        """
+        return self._container
+
+    def pull_model(self, model_name):
+        """
+        Pull a model from the Ollama server
+
+        Args:
+            model_name (str): Name of the model
+        """
+        self.exec(f"ollama pull {model_name}")
+
+    def commit_to_image(self, image_name):
+        """
+        Commit the current container to a new image
+
+        Args:
+            image_name (str): Name of the new image
+        """
+        docker_client = self.get_docker_client()
+        container = self.get_container()
+        existing_images = docker_client.client.images.list(name=image_name)
+        if not existing_images:
```

Comment:
If you don't add [this label](https://github.com/testcontainers/testcontainers-java/blob/main/modules/ollama/src/main/java/org/testcontainers/ollama/OllamaContainer.java#L70) Ryuk is going to remove the image. 
I missed that one when adding Ollama module for tc-node

Reply:
This label namespace is protected in the python SDK. I have this issue trying to use this label:
`ValueError: The org.testcontainers namespace is reserved for internal use`

Reply:
Hi @bfotzo, it should be something like `docker_client.client.containers.commit` (or something like this) instead of `container.commit`

Reply:
Hi @eddumelendez, that's what worked: `docker_client.client.containers.get(self.id).commit(repository=image_name)`following your recommendation

Reply:
Hi @eddumelendez, that's what worked following you recommendation:
```python
docker_client.client.containers.get(self.id).commit(repository=image_name)
```
`docker_client.client.containers` does not have the method commit directly

Reply:
> If you don't add [this label](https://github.com/testcontainers/testcontainers-java/blob/main/modules/ollama/src/main/java/org/testcontainers/ollama/OllamaContainer.java#L70) Ryuk is going to remove the image. I missed that one when adding Ollama module for tc-node

I fixed it, but had to change the fact that internal labels was not supposed to be changed.

Reply:
@bricefotzo the label shouldn't be set for the container but only when performing the `commit` operation. See https://docs.docker.com/engine/api/v1.45/#tag/Image/operation/ImageCommit

We have done something similar in Testcontainers for NodeJS, see https://github.com/testcontainers/testcontainers-node/pull/788/files#diff-212b3c0d5e84a8eb1a96cf9d0614e6d92b00feba22e9223ad86e4a66a7798d41R54

where:
- repo is a query param
- Labels is the request body

IDK if this is similar in python but we can try :D

Reply:
@eddumelendez, that's what I also wanted to do first but didn't find in the python sdk of docker Labels assignement. But I thing laybe using the parameter conf or changes, we can achieve that. 

Let me check and try

Reply:
@eddumelendez , that's what I also wanted to do first but didn't find in the python sdk of docker Labels assignement. But I thing laybe using the parameter conf or changes, we can achieve that.

Let me check and try

Reply:
Got it!
That's the way to do in python 
```python
docker_client.client.containers.get(self.id).commit(repository=image_name, conf={"Labels": {"org.testcontainers.session-id": ""}})
```
Thanks a lot @eddumelendez !